llvm/lib/Target/WebAssembly/AsmParser/WebAssemblyAsmParser.cpp

```diff
@@ -1002,7 +1004,8 @@ class WebAssemblyAsmParser final : public MCTargetAsmParser {
       auto *Signature = Ctx.createWasmSignature();
       if (parseSignature(Signature))
         return ParseStatus::Failure;
-      TC.funcDecl(*Signature);
+      if (CurrentState == FunctionStart)
+        TC.funcDecl(*Signature);
```

Comment:
`.functype` directive does not only happen at the start of a function but also just to declare a function type that does not have a definition associated with it. But we used to set the current function's signature whenever we parsed a `.functype`. This resulted (incorrectly) in calling `funcDecl` twice, and pushing to `BlockInfoStack` twice.